runtime/binding-grpc/src/main/java/io/aklivity/zilla/runtime/binding/grpc/internal/config/GrpcBindingConfig.java

```diff
@@ -174,13 +171,19 @@ private GrpcMethodConfig resolveMethod(
             .orElse(null);
     }
 
-
-    private Stream<GrpcProtobufConfig> resolveCatalogs()
+    private Collection<GrpcProtobufConfig> resolveProtobufs()
     {
-        for (CatalogedConfig catalog : catalogs)
+        ObjectHashSet<CatalogedConfig>.ObjectIterator catalogIterator = catalogs.iterator();
+
+        while (catalogIterator.hasNext())
         {
-            for (SchemaConfig catalogSchema : catalog.schemas)
+            CatalogedConfig catalog = catalogIterator.nextValue();
+            ObjectHashSet<SchemaConfig>.ObjectIterator schemaIterator = catalog.schemas.iterator();
+
+            while (schemaIterator.hasNext())
```

Comment:
If we're using `ObjectHashSet` with reusable iterator, then `for ( ... : ...)` syntax is fine, it will pick up the iterator implicitly.